### DISCUSSION TITLE
Removing Trailing Spaces from Registry Results

### DISCUSSION THREAD

**Discussion by RubicBG** [moudey/Shell#241] [Q&A]
I am attempting to retrieve stored settings from the registry. However, when I use the Registry functions, the results I receive contain a space at the end. The 'str.trim()' function fails to remove it, as it is not a typical space. Currently, I am using 'str.left(reg_data, str.len(reg_data)-1)' to remove the space, and it is functioning correctly. Are there any more efficient solutions available?

**Comment by moudey** [comment]
Does the space at the end of the value already exist in the Windows registry?

**Comment by RubicBG** [comment]
Here is part of reg export:

```
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\WinRAR\Interface\Themes]
"ActivePath"="POPCOL_96x96"

[HKEY_CURRENT_USER\Software\WinRAR\Profiles\0]
"Name"="Default Profile"

[HKEY_CURRENT_USER\Software\WinRAR\Profiles\1]
"Name"="Create e-mail attachment"

[HKEY_LOCAL_MACHINE\Software\WinRAR]
"exe64"="C:\\Program Files\\WinRar\\WinRAR.exe"
```

and from a dynamic .nss file, here's a var {} tail:


```
var 	
	{
	reg_winrar_exe64 = reg(reg.lm, 'Software\WinRAR', 'exe64') 
	reg_winrar_theme = reg(reg.cu, 'Software\WinRAR\Interface\Themes', 'ActivePath')
	
	path_winrar_theme = path.combine(sys.appdata, 'WinRAR\Themes', str.left(reg_winrar_theme, str.len(reg_winrar_theme)-1)) 
	
	reg_Prof0_tmp = reg(reg.cu, 'Software\WinRAR\Profiles\0','Name')
	reg_Prof0 = str.left(reg_Prof0_tmp, str.len(reg_Prof0_tmp)-1)
	}
```

At first, I didn't notice it because reg_winrar_exe64 didn't require any modifications, and the argument needed a space.

**Reply by moudey** [reply]
There is a bug in the `reg` function and it has been fixed in the latest [build](https://nilesoft.org/download/shell/debug.zip)

[New functions for handling the Registry](https://nilesoft.org/blog/new-functions-for-handling-the-registry)

**Comment by RubicBG** [comment]
Thank you for sending the latest build of the program. However, I'm encountering some issues with it.

Firstly, I've noticed that the tooltip doesn't appear anywhere, even in the Terminal menu, even when I test it with only the package and no other additional changes from me.

Secondly, after every restart of the explorer, everything works except the tooltip, and the additional menu on the taskbar remains visible. However, after I right-click on a file/folder/background in explorer (not on Desktop, only in explorer), the additional menu disappears completely and permanently, and it's only fixed by restarting the explorer.

I've tried removing the imports one by one to see if any of them cause the problem, but the condition remained the same. I also tried running the program in Windows Sandbox, and it worked without any issues. Additionally, I installed the old version using 'winget install nilesoft.shell,' and everything worked fine.

I'm not sure how to diagnose the issue, but I suspect that it may be conflicting with another program that's running on my computer.

Please let me know if there's anything I can do to help identify and resolve these issues.

**Reply by moudey** [reply]
Tip are disabled by default. You can activate them by adding this command to the `settings` section:
```
tip.enabled=true
```

**Reply by RubicBG** [reply]
i tried before with 'tip.enabled=true' in both shell.nss and theme.nss without success, but with this tip, 'settings.tip.enabled=true' works without fail. thanks.

about the disappearing taskbar menu, the problem is it's still there. if you have an idea for the future - write me back
